doc/nrf/ug_bt_mesh_vendor_model.rst

```diff
@@ -0,0 +1,422 @@
+The other arguments are set to NULL as optional.
+The next sections describe how and when to use these arguments.
+
+Note that vendor models are passed in the second parameter of :c:macro:`BT_MESH_ELEM`, when defining the node composition data:
+
+.. code-block:: c
+
+    static struct bt_mesh_elem elements[] = {
+        BT_MESH_ELEM(
+            1,
+            BT_MESH_MODEL_LIST(BT_MESH_MODEL_CFG_SRV(&cfg_srv),
+                               BT_MESH_MODEL_HEALTH_SRV(&health_srv,
+                                                        &health_pub)),
+            BT_MESH_MODEL_LIST(BT_MESH_MODEL_VND_CB(YOUR_COMPANY_ID,
+                                                    YOUR_MODEL_ID,
+                                                    BT_MESH_MODEL_NO_OPS,
+                                                    NULL,
+                                                    NULL,
+                                                    NULL))
+        ),
+    };
+
+Receiving messages
+==================
+
+If you want your model to receive messages, you need to create an opcode list.
+It defines a list of messages that your model will receive.
+To create the opcode list, initialize an array of :c:struct:`bt_mesh_model_op` type with the following required parameters:
+
+1. Message opcode, :c:member:`bt_mesh_model_op.opcode`, to register a message to be received by the model.
+#. Minimal message length, :c:member:`bt_mesh_model_op.min_len`, that prevents the model from receiving messages shorter than the specified value.
+#. Message handler, :c:member:`bt_mesh_model_op.func`, which is used to process the received message.
+
+Note that the last element in the opcode list is always :c:macro:`BT_MESH_MODEL_OP_END`:
+
+.. code-block:: c
+
+    static void handle_message_set(struct bt_mesh_model *model,
+                                   struct bt_mesh_msg_ctx *ctx,
+                                   struct net_buf_simple *buf)
+    {
+        // Message handler code
+    }
+
+    static void handle_message_ack(struct bt_mesh_model *model,
+                                   struct bt_mesh_msg_ctx *ctx,
+                                   struct net_buf_simple *buf)
+    {
+        // Message handler code
+    }
+
+    static void handle_message_status(struct bt_mesh_model *model,
+                                      struct bt_mesh_msg_ctx *ctx,
+                                      struct net_buf_simple *buf)
+    {
+        // Message handler code
+    }
+
+    const struct bt_mesh_model_op _opcode_list[] = {
+        { MESSAGE_SET_OPCODE,    MESSAGE_SET_LEN,    handle_message_set },
+        { MESSAGE_ACK_OPCODE,    MESSAGE_ACK_LEN,    handle_message_ack },
+        { MESSAGE_STATUS_OPCODE, MESSAGE_STATUS_LEN, handle_message_status },
+        BT_MESH_MODEL_OP_END,
+    };
+
+To associate the opcode list with your model, use :c:macro:`BT_MESH_MODEL_VND_CB`.
+It will initialize the :c:member:`bt_mesh_model.op` field of the model context.
```

Comment:
I actually had a code snippet that showed `BT_MESH_MODEL_VND_CB` for each step of the guide, but then removed them, because I decided that it is too obvious. But now I'll add them :)